src/sys/unix.rs

```diff
@@ -659,6 +665,75 @@ impl SockAddr {
             None
         }
     }
+
+    /// Returns true if this address is an unnamed address from the `AF_UNIX` (AKA `AF_LOCAL`)
+    /// family (for local interprocess communication), false otherwise.
+    pub fn is_unnamed(&self) -> bool {
+        self.as_sockaddr_un()
+            .map(|storage| {
+                self.len() == offset_of_path(storage) as u32
+                    // On some non-linux platforms a zeroed path is returned for unnamed.
+                    // Abstract addresses only exist on Linux.
+                    || (cfg!(not(any(target_os = "linux", target_os = "android")))
```

Comment:
Added fuchsia in https://github.com/rust-lang/socket2/pull/403/commits/8375cea06dc3ca108bda593ff6b8d87a82666d8c, although I couldn't really find any source that says whether Fuchsia has abstract-name unix sockets, and the existing code here did not include Fuchsia in platforms that have them:

https://github.com/rust-lang/socket2/blob/18a883653752116a10ec78f7d36f8d956e4c2e7c/tests/socket.rs#L150-L151

And neither does the standard library:
https://doc.rust-lang.org/src/std/os/unix/net/addr.rs.html#236

Reply:
@tamird can you confirm (or deny) that Fuchsia has abstract namespaces?

Reply:
Odd that `AF_UNIX` is defined for Fuchsia as the check passes.

Reply:
Yep, it's defined, but not implemented.

Reply:
@tamird is right, Fuchsia defines `AF_UNIX` but we do not (and have no plans to) implement unix domain sockets. 

We have `AF_UNIX` defined because its found in [musl](https://cs.opensource.google/fuchsia/fuchsia/+/main:zircon/third_party/ulib/musl/;drc=58986a79f3173fa3bb383f336df3828ca1f3d22d) "an MIT-licensed implementation of the standard C library targetting the Linux syscall API" that Fuchsia uses. 

Note including the unix address family on Fuchsia seems like the best path for now (please leave a comment mentioning this context so future changes do not accidentally add it).

Reply:
Thanks! mind linking this thread (https://github.com/rust-lang/socket2/pull/403/commits/e78bbd942af997b813ea6a860fa86993f13d05f1#r1123557978) in the comment to leave breadcrumbs?